src/include/snapper/dialogs.rb

```diff
@@ -343,33 +261,30 @@ def CreateSnapshotPopup(pre_snapshots)
       created
     end
 
-
+    
     # Popup for deleting existing snapshot
     # @return true if snapshot was deleted
-    def DeleteSnapshotPopup(snapshot)
-      snapshot = deep_copy(snapshot)
-      num = Ops.get_integer(snapshot, "num", 0)
-      pre_num = Ops.get_integer(snapshot, "pre_num", num)
-      type = Ops.get_symbol(snapshot, "type", :none)
-
-      if type != :POST
-
-        # yes/no popup question
-        if Popup.YesNo(_("Really delete snapshot %{num}?") % { :num => num })
-          return Snapper.DeleteSnapshot([ num ])
+    def DeleteSnapshotPopup(snapshots)
+      snaps = []
+      post_snaps = []
+      snapshots.map do |s|
```

Comment:
wrong usage. For this case `each` is better. If result is not two maps, then also `each_with_object` can be used.